packages/build/src/plugins_core/edge_functions/index.ts

```diff
@@ -104,18 +104,26 @@ const logFunctions = async ({
   logs,
   srcDirectory: userFunctionsSrc,
   srcPath,
-}) => {
-  const [userFunctions, internalFunctions] = await Promise.all([find([srcPath]), find([internalSrcPath])])
-  const userFunctionsSrcExists = await pathExists(srcPath)
-  const internalFunctionsSrc = internalSrcDirectory
+}: {
+  internalSrcDirectory: string
+  internalSrcPath: string
+  logs: any
+  srcDirectory: string
+  srcPath?: string
+}): Promise<void> => {
+  const [userFunctionsSrcExists, userFunctions, internalFunctions] = await Promise.all([
+    srcPath ? pathExists(srcPath) : Promise.resolve(false),
+    srcPath ? find([srcPath]) : Promise.resolve([]),
+    find([internalSrcPath]),
+  ])
```

Comment:
I added ` pathExists(srcPath)` also to the `Promise.all` so it also runs in parallel.

Reply:
The `srcPath ?` changes are because of TypeScript and srcPath being optional

Reply:
nit: I find this segment a bit hard to read. Having a destructured array with ternary expressions inside is a lot. It would be great if we could find a way to make this a bit nicer. Totally fine to do in a follow-up though.